storage/innobase/handler/ha_innodb.cc

```diff
@@ -12591,7 +12602,8 @@ bool create_table_info_t::create_option_tablespace_is_valid() {
 
   if (!m_use_shared_space) {
     if (!m_use_file_per_table) {
-      if (m_create_info->encrypt_type.str != nullptr && is_temp) {
+      if (m_create_info->encrypt_type.str != nullptr &&
+          m_create_info->explicit_encryption && is_temp) {
```

Comment:
How did this work before?
This change is to error out "CREATE TEMPORARY TABLE t1(a INT) ENCRYPTION=N" ?
Do we really need the explicit_encryption check here?

Reply:
agreed, not needed, removed.

Reply:
Correction - this is needed. CREATE TEMPORARY LIKE needs to retain encryption clause - even if it is N. Upstream did not care about the ENCRYPTION clause. We retain it. Thus here it will be not empty for implicit N (encrypt_type.str != nullptr ). We should fail only when ENCRYPTION was explicitly specified. Thus I need this check for it here.